subprojects/launcher/src/main/java/org/gradle/launcher/exec/InProcessBuildActionExecuter.java

```diff
@@ -42,6 +46,9 @@ public Object execute(BuildAction action, BuildRequestContext buildRequestContex
             RootBuildLifecycleListener buildLifecycleListener = contextServices.get(ListenerManager.class).getBroadcaster(RootBuildLifecycleListener.class);
             buildLifecycleListener.afterStart();
             try {
+                if (!JavaVersion.current().isJava8Compatible() && (actionParameters.getLogLevel() != QUIET && actionParameters.getLogLevel() != ERROR)) {
```

Comment:
:x: I believe this code shouldn't exist in this place. The `InProcessBuildActionExecuter` represents the mechanic to execute Gradle builds as part of an integration test. The deprecation message should come from Gradle itself and not the testing infrastructure.

Reply:
I agree with you, but if I remove the deprecation warning here, [this test](https://github.com/gradle/gradle/blob/3512ecab86759b8e203b96a609cb607b98c7f0f0/subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/ToolingApiDeprecatedBuildJvmCrossVersionSpec.groovy#L61) will fail on jdk7. My understanding is: if a jdk 7 JVM (`client`) connects to a jdk 7 JVM (`build`) and do some build, the warning should appear in the build's output. However, in this case, `client` JVM's deprecation warning has already appeared during its startup, so it wouldn't appear again (`DeprecationLogger`), and `build` JVM has no chance to run the version checking code. That's why it need to be here. 

Please let me know if I didn't make myself clear or I didn't understand the code correctly.

Reply:
I don't think so... it looks like all build operations will pass this code (and I'm inspired by [here](https://github.com/gradle/gradle/blob/ec877ed3bfc83e6905985da13cf0507dd8f05af4/subprojects/launcher/src/main/java/org/gradle/launcher/exec/InProcessBuildActionExecuter.java#L44)).

After struggling with failed tests, I found that deprecation warnings should be here so that bootstrap JVM and build JVM wouldn't generate duplicate warnings.